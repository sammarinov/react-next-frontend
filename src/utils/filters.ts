import { IFeatured, IArticle } from "@Redux/app/type";

export const filterFeatured = (featured: Array<IFeatured>, limit: number = -1) => {
    featured.sort((a, b) => {
        if (a.topFeatured && !b.topFeatured) return 1;
        if (!a.topFeatured && b.topFeatured) return -1;
        return parseInt(a.price) - parseInt(b.price);
    })

    if (limit > 0) return featured.slice(0, limit);
    return featured;
}

export const filterArticles = (articles: Array<IArticle>, limit: number = -1) => {
    articles.sort((a, b) => {
        return Date.parse(a.created) - Date.parse(b.created);
    })

    if (limit > 0) return articles.slice(0, limit);
    return articles;
}