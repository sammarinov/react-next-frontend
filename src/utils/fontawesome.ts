import { library } from '@fortawesome/fontawesome-svg-core';

import {
    faArrowRight
} from '@fortawesome/free-solid-svg-icons';

import {
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
} from '@fortawesome/free-brands-svg-icons';

library.add(
    faArrowRight,
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
)