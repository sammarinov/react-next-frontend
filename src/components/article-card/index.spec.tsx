import * as React from 'react';

import { render } from '@Test/utils';
import ArticleCard from '.';

describe('ArticleCard', () => {
    it('should match snapshot', () => {
        const { container } = render(<ArticleCard item={
            {
                "id": 1,
                "caption": "About Mickey Mouse",
                "description": "Micke Mouse is a cartoon character and the mascot of The Walt Disney Company. He was created by Walt Disney and Ub Iwerks at the Walt Disney Studios in 1928. An anthropomorphic mouse who typically wears red shorts, large yellow shoes, and white gloves, Mickey is one of the world's most recognizable characters.",
                "picture": "mickey.png",
                "created": "1984-08-05T08:15:30-05:00",
                "author": "Jummy"
            }
        } />);

        expect(container).toMatchSnapshot();
    });
});
