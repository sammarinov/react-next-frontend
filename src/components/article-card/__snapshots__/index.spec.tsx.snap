// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should match snapshot 1`] = `
<div>
  <div
    class=" article-card"
  >
    <img
      class="article-card-image"
      src="images/mickey.png"
    />
    <div
      class="article-card-caption"
    >
      About Mickey Mouse
    </div>
    <div
      class="article-card-info"
    >
      Aug 05, 1984
       by 
      Jummy
    </div>
    <div
      class="article-card-description"
    >
      Micke Mouse is a cartoon character and the mascot of The Walt Disney Company. He was created by Walt Disney and Ub Iwerks at the Walt Disney Studios in 1928. An anthropomorphic mouse who typically wears red shorts, large yellow shoes, and white gloves, Mickey is one of the world's most recognizable characters.
    </div>
  </div>
</div>
`;
